Improve tile editor selection appearance

KoBeWi opened the pull request:
Resolves #49371
![godot windows tools 64_y19wagA4qp](https://user-images.githubusercontent.com/2223172/167313414-a162cc55-c4a6-4230-81e2-e1582c8ac2fb.gif)

I added a static method that draws a ninepatch texture rect and unified all rects to use this method. Unfortunately it doesn't look good with small tiles, so some alternative is probably needed.
![image](https://user-images.githubusercontent.com/2223172/167313463-ad7bba90-dcf3-4c1a-abe6-9c995ee04d38.png)

KoBeWi committed:
Improve tile editor selection appearance

akien-mga commented:
> I added a static method that draws a ninepatch texture rect and unified all rects to use this method. Unfortunately it doesn't look good with small tiles, so some alternative is probably needed.

If it's drawn as an overlay, can't it have a higher resolution than the tile and thus always the same thickness? This would also prevent having it change thickness when zooming in and out the palette (assuming that's supported). I assume something similar is done/should be done for the circular handles shown at the 8 expand points.

groud commented:
I am ok with the idea, but indeed, the current situation is not acceptable with small tiles.
The width of the rect should be kept independent of the zoom level.

If it is too complicated to implement, we would need to reduce the width from three pixel to one pixel, and maybe use a black & white dashed pattern?

KoBeWi commented:
Rebased, but needs more work. The texture broke for some reason (it's no longer using nearest filtering and I don't know how to force it). I added a commit where I attempted to use a rectangle, but this reminded me why I gave up that approach :/

akien-mga commented:
Thanks!
